client/src/lib/dependency-injection/withInjector.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+
+function withInjector(Component, dependencies, mapDependenciesToProps) {
+  const injectorContext = {
+    injector: React.PropTypes.shape({
+      get: React.PropTypes.func,
+    }),
+  };
+
+  class Injectable extends React.Component {
+    render() {
+      let props = {};
+      if (dependencies) {
+        if (!Array.isArray(dependencies)) {
+          throw new Error(`
+            withInjector() passed an argument for dependencies that is ${typeof dependencies}. 
+            Must be an array of named dependencies.
+          `);
+          return;
```

Comment:
this `return` is unreachable.
it's also an inconsistent return value it should be `return null;` if it were still around ;)